src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -158,20 +158,18 @@ private void Add(string path, object value, T objectToApplyTo, Operation<T> oper
                         else
                         {
                             throw new JsonPatchException<T>(operationToReport,
-                                string.Format("Patch failed: provided path is invalid for array property type at " +
-                                    "location path: {0}: position larger than array size",
-                                    path, 422),
-                                objectToApplyTo);
+                                Resources.InvalidIndexForArrayProperty(operationToReport.op, path),
```

Comment:
Did the resx generation change recently? This should be `Resources.FormatInvalidIndexForArrayProperty`